src/ast_convenience.ml

```diff
@@ -81,7 +90,7 @@ let sequence ?loc ?attrs = function
   | hd :: tl -> List.fold_left (fun e1 e2 -> Exp.sequence ?loc ?attrs e1 e2) hd tl
 
 let pvar ?(loc = !default_loc) ?attrs s = Pat.var ~loc ?attrs (mkloc s loc)
-let pconstr ?loc ?attrs s args = Pat.construct ?loc ?attrs (lid ?loc s) (may_tuple ?loc Pat.tuple args)
+let pconstr ?loc ?attrs s args = Pat.construct ?loc ?attrs (lid ?loc s) (may_pat_tuple ?loc Pat.tuple args)
```

Comment:
It seems to me that it would be better to add an `?existentials` argument to ensure that all constructors patterns can be built by `pconstr`.

Reply:
To me it looks like this can be done later if needed. I prefer not to change the interface right now as to me `ppx_tools` is maintained for improved backward compatibility and not really for new features. `ppxlib` does the job much better and is the preferred path by the community